.github/workflows/build-test-release.yml

```diff
@@ -0,0 +1,208 @@
+
+  sonar-cloud:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install in develop mode
+        run: |
+          pip install -e .[dev]
+
+      - name: Test and Coverage
+        run: |
+          pytest --cov-report=xml:coverage.xml --cov-fail-under=0
+
+          # Fix relative paths in coverage file
+          # Known bug: https://community.sonarsource.com/t/sonar-on-github-actions-with-python-coverage-source-issue/36057
+          sed -i 's@/home/runner/work/power-grid-model-io/power-grid-model-io@/github/workspace@g' coverage.xml
+
+      - name: SonarCloud Scan
+        if: ${{ (github.event_name == 'push') || (github.event.pull_request.head.repo.owner.login == 'PowerGridModel') }}
+        uses: SonarSource/sonarqube-scan-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  unit-tests:
+    needs: build-python
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python: ["3.11", "3.12"]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup Python ${{ matrix.python }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Load built wheel file
+        uses: actions/download-artifact@v4
+        with:
+          name: power-grid-model-io
+          path: wheelhouse/
+
+      - name: Install built wheel file
+        run: pip install power-grid-model-io[dev]==${{ needs.build-python.outputs.version }} --find-links=wheelhouse
+
+      - name: Unit test and coverage
+        run: pytest --verbose
+
+  validation-tests:
+    needs: build-python
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python: ["3.11", "3.12"]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup Python ${{ matrix.python }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Load built wheel file
+        uses: actions/download-artifact@v4
+        with:
+          name: power-grid-model-io
+          path: wheelhouse/
+
+      - name: Install built wheel file
+        run: pip install power-grid-model-io[dev]==${{ needs.build-python.outputs.version }} --find-links=wheelhouse
+
+      - name: Validation tests
+        run: pytest tests/validation --no-cov --verbose
+
+  github-release:
```

Comment:
Shall we move `github-release` to ci.yml as well? 
I don't understand the reasoning behind running this with `inputs.create_release`=false. 

I ask because same is done in PGM repo.

Reply:
> Shall we move github-release to ci.yml as well?

we want to homogenize as much as possible between PGM-IO and PGM.

> I don't understand the reasoning behind running this with inputs.create_release=false.

i think that that is a topic for a whole other discussion. Let's first focus on making it work again